hashes/examples/hash-a-file.rs

```diff
@@ -0,0 +1,22 @@
+//! Use `bitcoin_hashes` to sha256 hash the contents of a file.
+
+use std::io::{self, BufRead};
+
+use bitcoin_hashes::prelude::*;
+use bitcoin_hashes::sha256;
+
+const FILE: &str = "a mock text file \n a line of text \n and another one \n";
+
+fn main() {
+    // This would usually be BufReader::new(File::open(path)?);
+    let reader = io::Cursor::new(&FILE);
+    let mut engine = sha256::engine();
+
+    for line in reader.lines() {
+        let line = line.expect("line read failed");
+        engine.input(line.as_bytes());
+    }
+    let digest = engine.digest();
```

Comment:
I copied the function name from a python library, not sure if its good to open the hash/digest can of worms.

Reply:
I kinda think this function should be called `finalize` or `extract`, which avoids worrying about what to name its output.

Reply:
`RustCrpto` uses "finish" (IIRC). I kinda `extract` now you mention it.

Reply:
Noting that @Kixunil used `finalize` in #1555, name could be bikeshedded, although I don't care strongly either way.

- `extract` is one character less
- both are equally as expressive IMO

Reply:
IIRC I've seen `finalize` in multiple other libs so this makes most sense to me.

Reply:
Cool, I'm good with `finalize`.